runner-review.sh

```diff
@@ -0,0 +1,333 @@
+while true; do
+    case "$1" in
+	
+	    -c)
+	    # Getting & store the execution count/times
+            export COUNT="$2"
+            shift 2
+            ;;
+			
+        --failed-count)
+            # Getting & store the failed execution count
+            export FAIL_COUNT="$2"
+            shift 2
+            ;;
+			
+        --debug)
+	    # Setting the debug options
+            export DEBUG="set -xvf"
+            shift
+            ;;
+			
+        --net-trace)
+	    # trigger tcpdump to generate network traffic : we used - /usr/sbin/tcpdump
+            export NET_TRACE="Y"
+            shift
+            ;;
+        --sys-trace)
+	    # triggering OPVA Extract  utility to generate system resources measured matrices : we used /opt/perf/bin/extract
+            export SYS_TRACE="Y"
+            shift
+            ;;
+			
+        --call-trace)
+	    # trigger the strace to generate call trace - we used /usr/bin/strace 
+            export CALL_TRACE="Y"
+            shift
+            ;;
+			
+        --log-trace)
+	    # trigger the log trace by printing the stdout & stderr to separate log files 
+            export LOG_TRACE="Y"
+            export LOG_OUT="${OUTPUT_DIR}/log_trace_`${TIME_SIGNUTURE}`.out"
+            export LOG_ERR="${OUTPUT_DIR}/log_trace_`${TIME_SIGNUTURE}`.err"
+            shift
+            ;;
+
+
+        --)
+		# once we arrived to -- this end of arguments parsing 
+            shift
+            break
+            ;;
+        *)
+		# when the input from OPTS no matching any of above , then it means the input not valid/or something wrong happened 
+            usage
+            ;;
+    esac
+done
+
+
+# enable/disable the debug option depend on the input options (if --debug was passed)
+${DEBUG}
+
+
+
+
+# handle non-option arguments , check if the input argument(not for the OPTS) mount is only 1 argument(one argument for the command parameter)
+if [[ $# -ne 1 ]]; then
+    usage
+fi
+
+
+# getting the command to run from command-line argument 
+COMMAND="$1"
+
+# declare array to store all the returned value for each command execution
+set -a list_rc
+
+
+
+
+
+
+
+# command executions loop - main invocations steps 
+# Test and check the executions time and allowed fails counters 
+while [[ ${execution_count} != ${COUNT} ]] && [[ ${failed_executions} != ${FAIL_COUNT} ]] ; do
+
+      export START_DATE=$(date +"%m/%d/%Y %H:%M:%S")
+      export CALL_TRACE_OUT="${OUTPUT_DIR}/call_trace_`${TIME_SIGNUTURE}`.out"
+
+
+       # if --net-trace enabled then trigger net traffic monitor
+       [[ ${NET_TRACE} ]] && exec_net_trace  1>/dev/null 2>&1  & 
+       wait_pids="${wait_pids} $!"
+
+
+
+      # --log-trace enable 
+      if [[ ${LOG_TRACE} ]] ; then 
+                                  
+          # --call-trace
+          if  [[ ${CALL_TRACE} ]] ; then 
+                  
+                  /usr/bin/strace -ttT -o ${CALL_TRACE_OUT} ${COMMAND} 1>>${LOG_OUT}  2>>${LOG_ERR} 
```

Comment:
You shouldn't have an assumption about the location of strace's binary, it does not have to be the same on all systems. 
Just use `strace ...` and let the user configure the `PATH` variable.

Reply:
Thanks for this comment .
i will take this in account , for next version of the script .